lib/rules/no-trailing-spaces.js

```diff
@@ -66,6 +71,22 @@ module.exports = {
             });
         }
 
+        /**
+         * Given a list of comment nodes, return the line numbers for those comments.
+         * @param {Array} comments An array of comment nodes.
+         * @returns {number[]} A array of line numbers containing comments.
+         */
+        function getCommentLineNumbers(comments) {
+            const lines = [];
+
+            comments.forEach(comment => {
+                for (let i = comment.loc.start.line; i <= comment.loc.end.line; i++) {
+                    lines.push(i);
```

Comment:
It looks like this will result in O(n) time lookups in the array because the `if` statement below calls `indexOf` on the resulting array. Can you use a `Set` instead of an array to store line numbers?